### PR TITLE
Metadata error validation

### PR DESCRIPTION
**Pull request by mthpvg**
### Planning
https://github.com/upfrontIO/livingdocs-planning/issues/1491

### Origin
 Gabriel wrote [here](https://github.com/upfrontIO/livingdocs-editor/pull/1753#pullrequestreview-79767555):
> write some documentation here https://docs.livingdocs.io/reference-docs/editor-configuration/metadata.html (at the end) how to create a new metadata form in a customizing project and provide an example with an inlined error message

### Remarks
It's a WIP, it could also tackle: [Document how to do custom metadata components](https://github.com/upfrontIO/livingdocs-planning/issues/1417)

### CODE REVIEW

**Comment by mthpvg** [comment]
I decided to go for a walkthrough/example of a fully customizable metadata component:
- server plugin
- editor's form
- editor's service

I think it tackles: [Document how to do custom metadata components](https://github.com/upfrontIO/livingdocs-planning/issues/1417)

I left out the error inlining on purpose, it will be in the boilerplates, for simplicity sake.

I **do not** have the best words when writing, I hope some parts are english-ok.

@gabrielhase can you review?

File: reference-docs/metadata/examples.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
+# Metadata examples
+
+
+
+## Fully customized metadata component
```
Comment (gabrielhase) [inline reference-docs/metadata/examples.md]: This should probably not be in the reference-docs folder, but the walkthroughs

File: reference-docs/metadata/examples.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+We would like to add a new metadata field, a `slug`, in the publish screen.
```
Comment (gabrielhase) [inline reference-docs/metadata/examples.md]: Please quickly explain what a `slug` is that might not be obvious to everybody.

File: reference-docs/metadata/examples.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+![Slug form metadata](./slug-metadata-form.png)
+
+### Server
+
+First we need to define a new property in our Elasticsearch's mappings.
```
Comment (gabrielhase) [inline reference-docs/metadata/examples.md]: `mappings` -> `mapping`

File: reference-docs/metadata/examples.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+  ]
+};
+
+```
+
+There is a new `metadata` key: `slug`. It has a custom server `plugin` set to `'bp-slug'`. We are going to see later on how to define this plugin.
+
+There is also a new object in the `metadataFormArrangement` array. It describes how the editor handles the `slug` metadata. It has a form: `'bp-slug-form'`, and a custom service: `'bpSlugService'`.
+
+The form is an Angular's component and the custom service refers to some business logic code.
```
Comment (gabrielhase) [inline reference-docs/metadata/examples.md]: `Angular's` -> `Angular`

File: reference-docs/metadata/examples.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+require('../plugins/metadata/bp-slug/form/bp-slug')(editorModule)
+
+liEditor.metadataServices
+  .register(
+    'bpSlugService',
+    require('../plugins/metadata/bp-slug/service')
+  )
+```
+
+### Comment
```
Comment (gabrielhase) [inline reference-docs/metadata/examples.md]: `Comment` -> `Boilerplate`

File: SUMMARY.md [hunk 6]
```diff
@@ -70,6 +70,8 @@
 
 * [Register a custom feature](walkthroughs/add_customizations.md)
 * [Define Metadata](concepts/metadata/metadata-examples.md)
+* [Metadata examples](reference-docs/metadata/examples.md)
+* [Customized metadata component](walkthroughs/metadata/examples.md)
```
Comment (mthpvg) [inline SUMMARY.md]: I was thinking about merging the example of this PR:
```
[Customized metadata component](walkthroughs/metadata/examples.md)
```
in:
```
[Metadata examples](reference-docs/metadata/examples.md)
```
You can check this document here: https://docs.livingdocs.io/concepts/metadata/metadata-examples.html

And finally maybe moving it from: `reference-docs` to `walkthroughs`.

@gabrielhase as you see fit.

**Pull request merged**